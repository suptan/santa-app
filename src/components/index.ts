export * from "./Button";
export * from "./Form";
export * from "./Footer";
export * from "./Header";
export * from "./Input";
export * from "./Loader";
export * from "./Wish";
