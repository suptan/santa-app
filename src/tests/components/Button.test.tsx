import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Button } from "@/components";

describe("<Button>", () => {
  it.each`
    color          | disabled | type
    ${"primary"}   | ${false} | ${"primary"}
    ${"primary"}   | ${true}  | ${"disabled primary"}
    ${"secondary"} | ${false} | ${"secondary"}
    ${"secondary"} | ${true}  | ${"disabled secondary"}
  `("should render as $type button", ({ color, disabled }) => {
    const { container } = render(
      <Button color={color} disabled={disabled}>
        ${color.charAt(0).toUpperCase() + color.slice(1)}
      </Button>
    );
    expect(container).toMatchSnapshot();
  });

  describe("onClick", () => {
    const mockClick = jest.fn();
    afterEach(() => {
      jest.clearAllMocks();
    });
    it("should execute callback function onClick", async () => {
      render(<Button onClick={mockClick}>text</Button>);
      const button = screen.getByRole("button");
      await userEvent.click(button);
      expect(mockClick).toHaveBeenCalledTimes(1);
    });

    it("should not allow click event when dislabed", async () => {
      render(
        <Button onClick={mockClick} disabled>
          text
        </Button>
      );
      const button = screen.getByRole("button");
      await userEvent.click(button);
      expect(mockClick).toHaveBeenCalledTimes(0);
    });
  });

  it("should render as link button when given href", () => {
    const { container } = render(
      <Button href="http://anywhere.gov">Link</Button>
    );
    expect(container).toMatchSnapshot();
  });
});
