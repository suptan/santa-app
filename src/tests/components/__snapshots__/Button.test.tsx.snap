// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> should render as disabled primary button 1`] = `
<div>
  <button
    class="py-2 px-4 rounded [&[disabled]]:bg-slate-200 bg-green-600 text-white hover:bg-green-500"
    disabled=""
    type="button"
  >
    $
    Primary
  </button>
</div>
`;

exports[`<Button> should render as disabled secondary button 1`] = `
<div>
  <button
    class="py-2 px-4 rounded [&[disabled]]:bg-slate-200 bg-red-600 text-white hover:bg-red-500"
    disabled=""
    type="button"
  >
    $
    Secondary
  </button>
</div>
`;

exports[`<Button> should render as link button when given href 1`] = `
<div>
  <a
    class="py-2 px-4 rounded [&[disabled]]:bg-slate-200"
    type="button"
  >
    Link
  </a>
</div>
`;

exports[`<Button> should render as primary button 1`] = `
<div>
  <button
    class="py-2 px-4 rounded [&[disabled]]:bg-slate-200 bg-green-600 text-white hover:bg-green-500"
    type="button"
  >
    $
    Primary
  </button>
</div>
`;

exports[`<Button> should render as secondary button 1`] = `
<div>
  <button
    class="py-2 px-4 rounded [&[disabled]]:bg-slate-200 bg-red-600 text-white hover:bg-red-500"
    type="button"
  >
    $
    Secondary
  </button>
</div>
`;
